frontend/hub/collections/CollectionDetails.tsx

```diff
@@ -55,20 +56,42 @@ import { HubItemsResponse } from '../useHubView';
 import { CollectionVersionSearch } from './Collection';
 import { useCollectionActions } from './hooks/useCollectionActions';
 import { useCollectionColumns } from './hooks/useCollectionColumns';
+import { PageSingleSelect } from './../../../framework/PageInputs/PageSingleSelect';
+import { CheckCircleIcon, ExclamationTriangleIcon } from '@patternfly/react-icons';
+import { Label } from '@patternfly/react-core';
+import { AwxError } from '../../awx/common/AwxError';
 
 export function CollectionDetails() {
   const { t } = useTranslation();
-  const params = useParams<{ name: string; namespace: string; repository: string }>();
-  const { data, refresh } = useGet<HubItemsResponse<CollectionVersionSearch>>(
-    hubAPI`/v3/plugin/ansible/search/collection-versions/?name=${params.name || ''}&namespace=${
-      params.namespace || ''
-    }&repository=${params.repository || ''} }`
+  const [searchParams] = useSearchParams();
+  const [collection, setCollection] = useState<CollectionVersionSearch | undefined>(undefined);
+  const itemActions = useCollectionActions(() => void refresh());
+
+  // load all collections versions belong to the repository
+  const { data, refresh, error, isLoading } = useGet<HubItemsResponse<CollectionVersionSearch>>(
+    hubAPI`/v3/plugin/ansible/search/collection-versions/?name=${
+      searchParams.get('name') || ''
+    }&namespace=${searchParams.get('namespace') || ''}&repository_name=${
+      searchParams.get('repository') || ''
```

Comment:
```suggestion
       collection?.repository.name || ''
```

maybe?

Reply:
Collection is selected collection. That would not work...

Reply:
Why not? The repo of the current collection sounds like the right thing to filter by when searching for other versions of the same collection in the same repo..

(but maybe I'm misunderstanding `searchParams`? I thought it was a query string param, but maybe it's a route param?)

Reply:
Never mind, I added version to search params, so now collection is loaded based on search params, not based on selection (selection changes search params version and collection is loaded then).